### ISSUE TITLE
java.lang.NullPointerException with cycling some maps

### ISSUE THREAD

**Issue by BeastPsycho** [twizmwazin/CardinalPGM#838]
[15:38:33] [Server thread/INFO]: [0;36;22mCycling to [0;36;1mTEST1[0;36;22m in [0;31;22m1[0;36;22m seconds[m
[15:38:34] [Server thread/INFO]: World matches/84e33212-235d-4334-9536-5985719caf33 is a duplicate of another world and has been prevented from loading. Please delete the uid.dat file from matches/84e33212-235d-4334-9536-5985719caf33's world directory if you want to be able to load the duplicate world.
[15:38:34] [Server thread/ERROR]: [CardinalPGM] Task #10050671 for Cardinal v1.0-SNAPSHOT-git-CardinalPGM-37548ff generated an exception
java.lang.NullPointerException
	at in.twizmwaz.cardinal.cycle.Cycle.run(Cycle.java:43) ~[?:?]
	at in.twizmwaz.cardinal.GameHandler.cycleAndMakeMatch(GameHandler.java:51) ~[?:?]
	at in.twizmwaz.cardinal.module.modules.cycleTimer.CycleTimerModule.run(CycleTimerModule.java:44) ~[?:?]
	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:53) ~[spigot2.jar:git-Bukkit-208668a]
	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:349) [spigot2.jar:git-Bukkit-208668a]
	at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:696) [spigot2.jar:git-Bukkit-208668a]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:335) [spigot2.jar:git-Bukkit-208668a]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:645) [spigot2.jar:git-Bukkit-208668a]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:555) [spigot2.jar:git-Bukkit-208668a]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_45]

**Comment by Alfalik** [comment]
Could you http://pastebin.com/ the entire log? Also, you need to use sportbukkit, not spigot. Here's a download https://repo.oc.tc/#nexus-search;quick~sportbukkit

**Comment by Prodigy4532** [comment]
Delete the Uuid.dat file.

**Comment by BeastPsycho** [comment]
I lost the full server log:

When do you leak the core or win the match (First TEST1 MAP), after playing, the plugin cycles to new map (TEST2), the next of rotation.txt, after the maps of that file are again "recycling" (again the plugin load TEST1) SOMETIMES the plugin cycles not to the same map because was not successfully unloaded when the plugin cycled from TEST1 map to TEST2 map, causing the "recycling of the map" be already finished, i stop the server, after restarting, the plugin deletes "matches/" with all map matches files and directory sucessfully.

Any way that this plugin unloads the cycled map sucessfully without restart the server?
I already saw the plugin source, i guees that you must add some like this for code:

GameHandler,Java
    public void cycleAndMakeMatch() {
        if (rotation.getNext().equals(cycle.getMap())) {
            rotation.move();
        }
        World oldMatchWorld = matchWorld == null ? null : matchWorld.get();
        cycle.run();
        if (match != null) match.unregisterModules();
        this.match = new Match(cycle.getUuid(), cycle.getMap());
        this.match.registerModules();
        Cardinal.getInstance().getLogger().info(this.match.getModules().size() + " modules loaded.");
        Bukkit.getServer().getPluginManager().callEvent(new CycleCompleteEvent(match));
        cycle = new Cycle(rotation.getNext(), UUID.randomUUID(), this);
        Bukkit.unloadWorld(oldMatchWorld, true);
         //Get mathc fle folder and Force the previous match map to be deleted after be unloaded"
    }

**Comment by twizmwazin** [comment]
The plugin is throwing an NPE when it references the world during cycling. This means the world is null, and couldn't be loaded. Make sure your map is packaged correctly and isn't corrupt. This doesn't appear to be a plugin issue, although we can leave this open until it is resolved.

**Comment by dentmaged** [comment]
@twizmwazin I believe this issue occurs when there are two maps of the same name.

**Comment by BeastPsycho** [comment]
@dentmaged No, the issue occurs when the plugin go to recycle the same map defined in rotation.txt after all maps are already played, i guess plugin fails to unload sucessfully the match, and the match folder it stays there and when the plugin will load the same map match, it uses the same data folder and its there when appears the NPE, when the plugin tried to load that map in the same data floder, by example:

rotation.txt defined maps: TestMap1 and TestMap2

Plugins load this:
TestMap1:
/matches/fa9130a8-a04d-40b6-8024-1507f787da83
After, the plugin load this map:
TestMap2:
/matches/cb9130a8-a04d-40b6-8024-1507b787da82
Again the server will load TestMap1, sometimes it generates othe UUID code map folder, but this load sucessully:
TestMap1:
/matches/ga9130a8-sd4d-20b6-8024-1507f787da84
Here or before, the plugin fails to load the "TestMap2" or the TestMap1 and it doesnot generates a new match map UUID data folder, with another name, "sometimes" and the server throws a NPE by a "duplicated map loading"
TestMap2:
/matches/cb9130a8-a04d-40b6-8024-1507b787da82

[Server thread/INFO]: World matches/cb9130a8-a04d-40b6-8024-1507b787da82 is a duplicate of another world and has been prevented from loading. Please delete the uid.dat file from matches/cb9130a8-a04d-40b6-8024-1507b787da82's world directory if you want to be able to load the duplicate world.
[15:38:34] [Server thread/ERROR]: [CardinalPGM] Task #10050671 for Cardinal v1.0-SNAPSHOT-git-CardinalPGM-37548ff generated an exception
java.lang.NullPointerException

I explained well, any way to fix?

**Comment by twizmwazin** [comment]
read your logs before posting them to issues, and package your maps correctly. This isn't a plugin issue.

**Comment by ryanw-se** [comment]
@twizmwazin If your expecting every single one of your users to be fluent in reading logs and understand the issue in this format, I must admit I'm a bit shocked. This could definitely be handled in a more user friendly fashion, besides spewing exceptions out.

**Comment by twizmwazin** [comment]
`[Server thread/INFO]: World matches/cb9130a8-a04d-40b6-8024-1507b787da82 is a duplicate of another world and has been prevented from loading. Please delete the uid.dat file from matches/cb9130a8-a04d-40b6-8024-1507b787da82's world directory if you want to be able to load the duplicate world.`

It's literally printed in plain english

**Comment by ryanw-se** [comment]
I'm talking about this part in particular that should and  can be remedied. ```[15:38:34] [Server thread/ERROR]: [CardinalPGM] Task #10050671 for Cardinal v1.0-SNAPSHOT-git-CardinalPGM-37548ff generated an exception
java.lang.NullPointerException
at in.twizmwaz.cardinal.cycle.Cycle.run(Cycle.java:43) ~[?:?]
at in.twizmwaz.cardinal.GameHandler.cycleAndMakeMatch(GameHandler.java:51) ~[?:?]
at in.twizmwaz.cardinal.module.modules.cycleTimer.CycleTimerModule.run(CycleTimerModule.java:44) ~[?:?]
at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:53) ~[spigot2.jar:git-Bukkit-208668a]
at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:349) [spigot2.jar:git-Bukkit-208668a]
at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:696) [spigot2.jar:git-Bukkit-208668a]
at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:335) [spigot2.jar:git-Bukkit-208668a]
at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:645) [spigot2.jar:git-Bukkit-208668a]
at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:555) [spigot2.jar:git-Bukkit-208668a]
at java.lang.Thread.run(Thread.java:745) [?:1.8.0_45]```
